packages/apps/src/microsoft/teams/apps/http_plugin.py

```diff
@@ -249,12 +252,15 @@ async def _handle_activity_request(self, request: Request) -> Any:
         response_future = asyncio.get_event_loop().create_future()
         self.pending[activity_id] = response_future
 
+        type_adapter: TypeAdapter[Activity] = TypeAdapter(Activity)
+        activity = type_adapter.validate_python(body)
+
         # Fire activity processing via callback
         if callable(self.on_activity_event):
```

Comment:
why are we checking for on_activity_event being a callable? we don't seem to use it here